packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ServiceClientBuilderTemplate.java

```diff
@@ -465,11 +471,33 @@ protected void writeSyncClientBuildMethod(AsyncSyncClient syncClient, AsyncSyncC
 
     protected void writeSyncClientBuildMethodFromInnerClient(AsyncSyncClient syncClient, JavaBlock function,
         String buildMethodName, boolean wrapServiceClient) {
+
+        boolean writeInstrumentation = !JavaSettings.getInstance().isAzureV1();
+
+        if (writeInstrumentation) {
+            function.line(
+                "HttpInstrumentationOptions localHttpInstrumentationOptions = this.httpInstrumentationOptions == null ? new HttpInstrumentationOptions() : this.httpInstrumentationOptions;");
+
+            boolean hasEndpoint = syncClient.getClientBuilder()
+                .getBuilderTraits()
+                .stream()
+                .anyMatch(trait -> trait instanceof EndpointTrait);
```

Comment:
Doesn't seem like this is triggering properly, `ApiKeyClientBuilder` in this PR has `EndpointTrait` but isn't calling `.setEndpoint` on the instrumentation options.

Reply:
great catch, fixed!